includes/enrolment/class-sensei-enrolment-job-scheduler.php

```diff
@@ -60,23 +60,49 @@ public function init() {
 	 *
 	 * @param int  $course_id        Course post ID.
 	 * @param bool $invalidated_only Recalculate just the results that have been invalidated (set to an empty string).
-	 * @param int  $batch_size       Batch size for the job. Null will use default batch size set by job handler.
 	 *
 	 * @return Sensei_Enrolment_Course_Calculation_Job Job object.
 	 */
-	public function start_course_calculation_job( $course_id, $invalidated_only, $batch_size = null ) {
+	public function start_course_calculation_job( $course_id, $invalidated_only ) {
 		$args = [
 			'course_id'        => $course_id,
 			'invalidated_only' => $invalidated_only,
-			'batch_size'       => $batch_size,
 		];
 
+		// Make sure all previous jobs for this course are stopped.
+		$this->cancel_course_calculation_job( $course_id );
```

Comment:
Since it's not meant to have more than two jobs for a single course running, should we move this to the start/setup method?

Reply:
I initially had it there, but didn't want the jobs to be aware of the Scheduler / create a funny dependency triangle.

Reply:
Cool let's avoid the dependency in `Sensei_Scheduler`